main/lsp/requests/code_action.cc

```diff
@@ -105,12 +326,33 @@ unique_ptr<ResponseMessage> CodeActionTask::runRequest(LSPTypecheckerInterface &
             auto action = make_unique<CodeAction>("Apply all Sorbet fixes for file");
             action->kind = kind;
             auto workspaceEdit = make_unique<WorkspaceEdit>();
-            workspaceEdit->documentChanges = getEdits(config, gs, allEdits);
+            workspaceEdit->documentChanges = getQuickfixEdits(config, gs, allEdits);
             action->edit = move(workspaceEdit);
             result.emplace_back(move(action));
         }
     }
 
+    if (config.opts.lspExtractMethodEnabled) {
+        auto queryResult = queryByLoc(typechecker, params->textDocument->uri, *params->range->start,
+                                      LSPMethod::TextDocumentCodeAction, false);
+
+        // Generate "Extract method" code actions only for method definitions
+        if (queryResult.error == nullptr) {
+            for (auto &resp : queryResult.responses) {
+                if (auto def = resp->isDefinition()) {
+                    if (def->symbol.isMethod()) {
+                        auto action = make_unique<CodeAction>("Extract method to module");
+                        action->kind = CodeActionKind::RefactorExtract;
+                        auto workspaceEdit = make_unique<WorkspaceEdit>();
+                        workspaceEdit->documentChanges = getExtractMethodEdits(config, gs, def, typechecker);
+                        action->edit = move(workspaceEdit);
+                        result.emplace_back(move(action));
```

Comment:
I haven't looked very closely at the LSP spec. I see that we're always constructing a workspaceEdit and emplacing it back, no matter what. But I think that there are certain cases where `getExtractMethodEdits` can fail to produce an edit.

Does VS Code get tricked if the list of actions is non-empty, but the action itself does not actually have an effect?